x/gamm/keeper/pool_test.go

```diff
@@ -316,3 +335,96 @@ func (suite *KeeperTestSuite) TestGetPoolAndPoke() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestSetStableSwapScalingFactors() {
+
+	
+	controllerPk := ed25519.GenPrivKey().PubKey()
+	controllerAddr := sdk.AccAddress(controllerPk.Address())
+
+	failPk := ed25519.GenPrivKey().PubKey()
+	failAddr := sdk.AccAddress(failPk.Address())
+
+	testcases := []struct {
+		name             string
+		poolId           uint64
+		scalingFactors   []uint64
+		sender           sdk.AccAddress
+		expError         error
+		isStableSwapPool bool
+	}{
+		{
+			name:             "Error: Pool does not exist",
+			poolId:           2,
+			scalingFactors:   []uint64{1, 1},
+			sender:           controllerAddr,
+			expError:         types.PoolDoesNotExistError{PoolId: defaultPoolId + 1},
+			isStableSwapPool: false,
+		},
+		{
+			name:             "Error: Pool id is not of type stableswap pool",
+			poolId:           1,
+			scalingFactors:   []uint64{1, 1},
+			sender:           controllerAddr,
+			expError:         fmt.Errorf("pool id 1 is not of type stableswap pool"),
+			isStableSwapPool: false,
+		},
+		{
+			name:             "Error: can not set scaling factors",
+			poolId:           1,
+			scalingFactors:   []uint64{1, 1},
+			sender:           failAddr,
+			expError:         types.ErrNotScalingFactorGovernor,
+			isStableSwapPool: true,
+		},
+		{
+			name:             "Valid case",
+			poolId:           1,
+			scalingFactors:   []uint64{1, 1},
+			sender:           controllerAddr,
+			isStableSwapPool: true,
+		},
+	}
+	for _, tc := range testcases {
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+			// Mint some assets to the accounts.
+			if tc.isStableSwapPool == true {
+				suite.FundAcc(tc.sender, defaultAcctFundsStableSwap)
+				poolId, err := suite.App.GAMMKeeper.CreatePool(
```

Comment:
We have helpers such as `PrepareBalancerPool` and `PrepareBasicStableswapPool`. Can we use them please?

Reply:
There are also `prepareCustom...` versions if you need to customize the pools